File: Tutorial4-1DependentComponent/src/main/java/com/example/tutorial4_1dependentcomponent/di/VehicleModule.java
package com.example.tutorial4_1dependentcomponent.di;

import com.example.tutorial4_1dependentcomponent.model.Motor;
import com.example.tutorial4_1dependentcomponent.model.Vehicle;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class VehicleModule {
    @Provides
    Vehicle provideVehicle(Motor motor) {
        return new Vehicle(motor);
    }
}

File: Tutorial4-1DependentComponent/src/main/java/com/example/tutorial4_1dependentcomponent/di/VehicleModule.kt
package com.example.tutorial4_1dependentcomponent.di

import com.example.tutorial4_1dependentcomponent.model.Motor
import com.example.tutorial4_1dependentcomponent.model.Vehicle
import dagger.Module
import dagger.Provides

@Module
class VehicleModule {

    @Provides
    fun provideVehicle(motor: Motor): Vehicle {
        return Vehicle(motor)
    }

}